google/cloud/aiplatform/featurestore/entity_type.py

```diff
@@ -1138,3 +1146,162 @@ def ingest_from_gcs(
+                entity_ids=entity_ids,
+                feature_selector=feature_selector,
+            )
+            streaming_read_feature_values_responses = [
+                response
+                for response in self._featurestore_online_client.streaming_read_feature_values(
+                    request=streaming_read_feature_values_request,
+                    metadata=request_metadata,
+                )
+            ]
+            header = streaming_read_feature_values_responses[0].header
+            entity_views = [
+                response.entity_view
+                for response in streaming_read_feature_values_responses[1:]
+            ]
+
+        feature_metadata = {
+            feature_descriptor.id: self.get_feature(
+                feature_descriptor.id
+            ).gca_resource.value_type
+            for feature_descriptor in header.feature_descriptors
+        }
+
+        return EntityType._construct_dataframe(
+            feature_metadata=feature_metadata, entity_views=entity_views,
+        )
+
+    @staticmethod
+    def _construct_dataframe(
+        feature_metadata: Dict[str, gca_feature.Feature.ValueType],
+        entity_views: List[
+            gca_featurestore_online_service.ReadFeatureValuesResponse.EntityView
+        ],
+    ) -> "pd.DataFrame":  # noqa: F821 - skip check for undefined name 'pd'
+        """Constructs a dataframe using the header and entity_views
+
+        Args:
+            feature_metadata (Dict[str, gca_feature.Feature.ValueType]):
+                Required. A dictionary of where the key is the feature id and the value is the feature value type.
+            entity_views (List[gca_featurestore_online_service.ReadFeatureValuesResponse.EntityView]):
+                Required. A list of Entity views with Feature values.
+                For each Entity view, it may be
+                the entity in the Featurestore if values for all
+                Features were requested, or a projection of the
+                entity in the Featurestore if values for only
+                some Features were requested.
+
+        Raises:
+            ImportError: If pandas is not installed when using this method.
+
+        Returns:
+            pd.DataFrame - entities feature values in DataFrame
+        )
+        """
+
+        try:
+            import numpy as np
+            import pandas as pd
+        except ImportError:
+            raise ImportError(
+                f"Numpy and/or Pandas are not installed. Please install numpy and/or pandas to use "
+                f"{EntityType._construct_dataframe.__name__}"
+            )
+
+        data = []
+        feature_ids = list(feature_metadata.keys())
+        for entity_view in entity_views:
+            entity_data = {"entity_id": entity_view.entity_id}
+            for feature_id, entity_view_data in zip(feature_ids, entity_view.data):
+                if entity_view_data._pb.HasField("value"):
```

Comment:
Added enforced None for entity_view_data proto withou the "value".

FYI, for data contains None, when the feature value type is "int64", pd will convert the dtype to np.dtype("float64"), with NaN instead of None. For other value types, dtype will be np.dtype("object") with None.